test/test_pipeline/components/data_preprocessing/test_data_preprocessing_text.py

```diff
@@ -0,0 +1,65 @@
+import unittest
+import numpy as np
+import pandas as pd
+from scipy import sparse
+
+import pytest
+
+from autosklearn.pipeline.components.data_preprocessing.text_encoding.bag_of_word_encoding import \
+    BagOfWordEncoder as BOW
+from autosklearn.pipeline.components.data_preprocessing.text_encoding.bag_of_word_encoding_distinct import \
+    BagOfWordEncoder as BOW_distinct
+
+
+class CategoricalPreprocessingPipelineTest(unittest.TestCase):
+
+    def test_fit_transform(self):
+        X = pd.DataFrame({"col1": ["Hello World",
```

Comment:
@eddiebergman would it make sense to directly start with `pytest`-style tests? In any case, this dataframe should be a class variable, right?

Reply:
For expediency, we can just test it in the same fashion

Reply:
Since it's repeated, it should exist outside the function scope. It can either exist as a class variable or even just be at the top of the file. I don't mind too much since all of these should eventually be changed to `pytest` style.

FYI for @mfeurer, this would be a wonderful example of `pytest-cases`. Making it into a case means it can be injected anywhere we need a text dataset.
```python
@case(tags=["dataset", "text"])
def case_simple_text_dataset():
    return ...
    
@parametrize_with_cases("data", filter=has_tag("dataset") & has_tag("text"))
def test_text_pipeline(data):
    ...
```